src/ops.rs

```diff
@@ -1376,3 +1381,77 @@ fn op_resources(
     },
   ))
 }
+
+fn op_run(
+  state: &Arc<IsolateState>,
+  base: &msg::Base,
+  data: &'static mut [u8],
+) -> Box<Op> {
+  let cmd_id = base.cmd_id();
+
+  // TODO: add a separate permission for child processes.
+  if let Err(e) = state.check_write("/") {
+    return odd_future(e);
+  }
+
+  assert_eq!(data.len(), 0);
+  let inner = base.inner_as_run().unwrap();
+  let argv = inner.argv().unwrap();
+  let dir = inner.dir();
+
+  let mut cmd = Command::new(argv.get(0));
+  (1..argv.len()).for_each(|i| {
+    let arg = argv.get(i);
+    cmd.arg(arg);
+  });
+  dir.map(|d| cmd.current_dir(d));
+
+  // TODO: add stdio etc.
+
+  let future = match cmd.status_async() {
+    Ok(v) => v,
+    Err(err) => {
+      return odd_future(err.into());
+    }
+  };
+  let future = future.map_err(DenoError::from).and_then(move |status| {
+    // TODO: rather than returning a future, make the process a resource so it
+    // can be interacted with (killed, piped into, etc.) before it exits.
```

Comment:
When this is changed to be a resource, will it be a sync op? Presumably it will return a rid ?  How will users wait for completion in this future API?

Reply:
I have made these changes. `deno.run()` is now sync and returns a `deno.Process` object (Is `Subprcoess` better?) which has an async method `process.status()` to wait for completion. `process.close()` must be called to clean up the resource.